statsmodels/base/tests/test_generic_methods.py

```diff
@@ -186,6 +187,175 @@ def test_predict_types(self):
+        assert_allclose(res1.pvalues[keep_index_p], res2.pvalues, rtol=1e-10)
+
+        if hasattr(res1, 'resid'):
+            # discrete models, Logit don't have `resid` yet
+            assert_allclose(res1.resid, res2.resid, rtol=1e-10)
+
+        ex = self.results.model.exog.mean(0)
+        predicted1 = res1.predict(ex)
+        predicted2 = res2.predict(ex[keep_index])
+        assert_allclose(predicted1, predicted2, rtol=1e-10)
+
+        ex = self.results.model.exog[:5]
+        predicted1 = res1.predict(ex)
+        predicted2 = res2.predict(ex[:, keep_index])
+        assert_allclose(predicted1, predicted2, rtol=1e-10)
+
+
+    def _get_constrained(self, keep_index, keep_index_p):
+        # override in some test classes, no fit_kwds yet, e.g. cov_type
+        mod2 = self.results.model
+        mod_cls = mod2.__class__
+        init_kwds = mod2._get_init_kwds()
+        mod = mod_cls(mod2.endog, mod2.exog[:, keep_index], **init_kwds)
+        if self.use_start_params:
+            res = mod.fit(start_params=self.results.params[keep_index_p])
+        else:
+            res = mod.fit()
+        return res
+
+
+    def test_zero_collinear(self):
+        # not completely generic yet
+        if (isinstance(self.results.model, (sm.GEE))):#, sm.OLS, sm.WLS))):
+            raise SkipTest
+
+        use_start_params = not isinstance(self.results.model,
+                                          (sm.RLM, sm.OLS, sm.WLS, sm.GLM))
+                                           # sm.NegativeBinomial))
+        self.use_start_params = use_start_params  # attach for _get_constrained
+        keep_index = list(range(self.results.model.exog.shape[1]))
+        # index for params might include extra params
+        keep_index_p = list(range(self.results.params.shape[0]))
+        drop_index = []
+        for i in drop_index:
+            del keep_index[i]
+            del keep_index_p[i]
+
+        keep_index_p = list(range(self.results.params.shape[0]))
+
+        #create collinear model
+        mod2 = self.results.model
+        mod_cls = mod2.__class__
+        init_kwds = mod2._get_init_kwds()
+        ex = np.column_stack((mod2.exog, mod2.exog))
+        mod = mod_cls(mod2.endog, ex, **init_kwds)
+
+        keep_index = list(range(self.results.model.exog.shape[1]))
+        keep_index_p = list(range(self.results.model.exog.shape[1]))
+        k_vars = ex.shape[1]
+        k_extra = 0
+        if hasattr(mod, 'k_extra') and mod.k_extra > 0:
+            keep_index_p += list(range(k_vars, k_vars + mod.k_extra))
+            k_extra = mod.k_extra
+
+        cov_types = ['nonrobust', 'HC0']
+
+        for cov_type in cov_types:
+            if cov_type != 'nonrobust' and (isinstance(self.results.model, (sm.RLM))):
+                raise SkipTest
+
+            if use_start_params:
+                start_params = np.zeros(k_vars + k_extra)
+                method =  self.results.mle_settings['optimizer']  #string not mutable
```

Comment:
here and in next line too many spaces after `=`.  Also comment `#string` --> `# string` (and possibly flesh that out since it isn't obvious why immutability is relevant